### ISSUE TITLE
Null Pointer Exception on decodeFrame

### ISSUE THREAD

**Issue by astine** [jcodec/jcodec#83]
Hello, I'm attempting to decode packets from an h264 stream in an mpeg-ts container. I'm getting a null pointer exception. 

```java
FileChannelWrapper bc = new FileChannelWrapper(new FileInputStream(file).getChannel());
Set<Integer> programs = MTSDemuxer.getPrograms(bc);
for (Integer program : programs) {
                MTSDemuxer mts = new MTSDemuxer(bc, program);
                for (MPEGDemuxer.MPEGDemuxerTrack track : mts.getTracks()) {
                    DemuxerTrackMeta meta = track.getMeta();
                    if(meta.getType() == DemuxerTrackMeta.Type.VIDEO) {
                        ByteBuffer bb = ByteBuffer.allocate(5000);
                        System.out.println(meta.getDimensions());
                        int[][] out = new int[][] {new int[10000000], new int[10000000], new int[10000000]};
                        Packet packet = track.nextFrame(bb);
                        ByteBuffer data = packet.getData();
                        VideoDecoder vd = JCodecUtil.detectDecoder(data);
                        System.out.println(vd);
                        data.rewind();
                        Picture pic = vd.decodeFrame(data, out);
//Throws: 
//error: java.lang.NullPointerException: null
//org.jcodec.codecs.h264.H264Decoder$FrameDecoder.init(H264Decoder.java:126)
//org.jcodec.codecs.h264.H264Decoder$FrameDecoder.decodeFrame(H264Decoder.java:87)
//org.jcodec.codecs.h264.H264Decoder.decodeFrame(H264Decoder.java:58)
//org.jcodec.codecs.h264.H264Decoder.decodeFrame(H264Decoder.java:41)
                    } else if (meta.getType() == DemuxerTrackMeta.Type.AUDIO) {
                        ByteBuffer bb = ByteBuffer.allocate(5000);
                        Packet packet = track.nextFrame(bb);
                    }
                }
            }
```

Am I missing something?

**Comment by svitvitskiy** [comment]
Could you post a file? I will look into it

**Comment by astine** [comment]
Here's an example of a file that gives me this problem:
http://theatticlight.net/output.ts
thanks for taking a look at this.

**Comment by svitvitskiy** [comment]
This would be a correct way to do it:
```java
SeekableByteChannel bc = null;
try {
    bc = NIOUtils.readableFileChannel(cmd.getArg(0));
    Set<Integer> programs = MTSDemuxer.getPrograms(bc);
    for (int program : programs) {
        System.out.println("Transcoding program " + String.format("%x", program));
        // Demuxer reads from a file, but only a specified program
        // so before going to the next program we need to rewind the file pointer
        bc.position(0);
        MTSDemuxer mts = new MTSDemuxer(bc, program);
        // We ignore all audio tracks, otherwise audio packets will just cache inside the
        // appropriate track objects and will cause memory overflow at some point
        for (MPEGDemuxerTrack track : mts.getAudioTracks()) {
            track.ignore();
        }
        List<? extends MPEGDemuxerTrack> videoTracks = mts.getVideoTracks();
        if (videoTracks.size() == 0)
            continue;
        MPEGDemuxerTrack videoTrack = videoTracks.remove(0);
        // We ignore all video tracks but the first. Since we are not going to pull packets
        // from any of the video tracks but first we are telling it to not cache the packets
        for (MPEGDemuxerTrack track : videoTracks) {
            track.ignore();
        }
        // 500k buffer for the raw frame, an AVC frame is anything from 500 bytes
        // to 100k. so 500k is a pretty safe bet
        ByteBuffer bb = ByteBuffer.allocate(500 << 10);
        // Since we don't know the resolution yet (before we've decoded at least one
        // frame allocate the maximum size of the buffer we can expect. MPEGTS doesn't
        // store video metadata, but it's possible to implement prober based on codec
        // specific headers.
        Picture tmp = Picture.create(1920, 1088, ColorSpace.YUV420);
        VideoDecoder vd = null;

        Packet packet;
        int i = 0;
        while ((packet = videoTrack.nextFrame(bb)) != null) {
            ByteBuffer data = packet.getData();
            if (vd == null)
                vd = JCodecUtil.detectDecoder(data.duplicate());
            Picture pic = vd.decodeFrame(data, tmp.getData());
            // As of JCodec 0.2 h264 decoder will return 'null' in case packet failed to
            // decode. For ex. when packet came before an appropriate SPS/PPS
            // metadata
            if (pic != null) {
                AWTUtil.savePicture(pic, "png", tildeExpand(format(cmd.getArg(1), i++)));
            }
        }
    }
} finally {
    NIOUtils.closeQuietly(bc);
}
```

**Comment by svitvitskiy** [comment]
Also, there's a bug in JCodec related to 16bit signed overflow. So from your test video only around 30 frames will process.
This commit fixes it: 233dc0d7a4af8e06c49ff1a8da2bf816f58baf5e
